src/test/java/com/suse/salt/netapi/calls/modules/TimezoneTest.java

```diff
@@ -0,0 +1,85 @@
+package com.suse.salt.netapi.calls.modules;
+
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import com.google.gson.JsonNull;
+import com.suse.salt.netapi.calls.LocalCall;
+import com.suse.salt.netapi.client.SaltClient;
+import com.suse.salt.netapi.client.impl.HttpAsyncClientImpl;
+import com.suse.salt.netapi.datatypes.AuthMethod;
+import com.suse.salt.netapi.datatypes.Token;
+import com.suse.salt.netapi.datatypes.target.MinionList;
+import com.suse.salt.netapi.errors.JsonParsingError;
+import com.suse.salt.netapi.results.Result;
+import com.suse.salt.netapi.utils.ClientUtils;
+import com.suse.salt.netapi.utils.TestUtils;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.util.Map;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.any;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlMatching;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Timezone module unit tests.
+ */
+public class TimezoneTest {
+
+    private static final int MOCK_HTTP_PORT = 8888;
+
+    private static final String JSON_NULL_RESPONSE = ClientUtils.streamToString(
+            TimezoneTest.class.getResourceAsStream(
+            "/modules/timezone/null_response.json"));
+
+    private static final String JSON_GETOFFESET_OK_RESPONSE = ClientUtils.streamToString(
+            TimezoneTest.class.getResourceAsStream(
+            "/modules/timezone/getOffset_ok.json"));
+
+    private SaltClient client;
+
+    @Rule
+    public WireMockRule wireMockRule = new WireMockRule(MOCK_HTTP_PORT);
+
+    static final AuthMethod AUTH = new AuthMethod(new Token());
+
+    @Before
+    public void init() {
+        URI uri = URI.create("http://localhost:" + Integer.toString(MOCK_HTTP_PORT));
+        client = new SaltClient(uri, new HttpAsyncClientImpl(TestUtils.defaultClient()));
+    }
+
+    @Test
+    public final void testGetOffset() {
+        // Test for successful null response
+        stubFor(any(urlMatching("/"))
+                .willReturn(aResponse()
+                .withStatus(HttpURLConnection.HTTP_OK)
+                .withHeader("Content-Type", "application/json")
+                .withBody(JSON_NULL_RESPONSE)));
+
+        LocalCall<String> call = Timezone.getOffset();
+        assertEquals("timezone.get_offset", call.getPayload().get("fun"));
+
+        Map<String, Result<String>> response = call.callSync(client,
+                new MinionList("minion1"), AUTH).toCompletableFuture().join();
+        assertEquals(JsonNull.INSTANCE,
+                ((JsonParsingError) response.get("minion1").error().get()).getJson());
+
+        // Test for successful responses
```

Comment:
I would suggest to test each case separately here, would you maybe want to move this second use case into a separate method?

Reply:
I agree with you, I think each case should be in a separated method and that was my first approach. 
The reason behind my change of approach was to keep the consistency with the rest of the project test, with one test method per function with several scenarios.
However in this case following the right way instead of consistency seems not to bring any harm to the future of the project, therefore I will  split the test into two as suggested.